builder/cloudstack/step_create_instance.go

```diff
@@ -65,7 +115,29 @@ func (s *stepCreateInstance) Run(state multistep.StateBag) multistep.StepAction
 	}
 
 	if config.UserData != "" {
-		ud, err := getUserData(config.UserData, config.HTTPGetOnly)
+		var httpPort uint = state.Get("http_port").(uint)
+		hostIp, err := hostIP()
+		if err != nil {
+			state.Put("error", err)
```

Comment:
I think that if you don't check anywhere in your builder for `state.Get("error")`, there is no use for putting it in. Unless this specific entry is one that is used by Packer core logic. 

@rickard-von-essen can you say anything about that?